Bugfix for L2 norm for AbstractLongDoubleVector

twolfe18 opened the pull request:
- used to return the squared L2 norm, now return the actual L2 norm

NOTE: I used Math.sqrt instead of another variant like FastMath because the cost of the sqrt should be small compared to iterating over the entire vector, and Java's Math.sqrt is "standard".

twolfe18 committed:
Bugfix for L2 norm for AbstractLongDoubleVector

- used to return the squared L2 norm, now return the actual L2 norm

mgormley commented:
Good catch. As is, your change breaks the build. TravisCI shows the following error on four files:
"error: incompatible types: possible lossy conversion from double to long"

Could you address this? It probably just needs some careful casting so that the IntLong, IntInt versions of Vector still work fine.

twolfe18 committed:
Upgraded tests to handle L2 norm fix

- Mutable(Long|Int|Short|Double|Float) now have a sqrt() method which returns the appropriate type

twolfe18 commented:
Sorry about that, I made that commit as I was running out the door. I had to change one test of the L2 norm to make it work for the integer-valued vector variants.

mgormley commented:
Looks great. Thanks again.
